src/bindings/event.ts

```diff
@@ -43,11 +48,21 @@ export const bindEvents = async (context: Context): Promise<void> => {
     // level: botConfig.log.level,
   };
 
-  logger = new GitHubLogger(options.app, botConfig.log.logEnvironment, botConfig.log.level, botConfig.log.retryLimit); // contributors will see logs in console while on development env
+  logger = new GitHubLogger(
+    options.app,
+    botConfig?.log?.logEnvironment ?? "development",
+    botConfig?.log?.level ?? Level.DEBUG,
+    botConfig?.log?.retryLimit ?? 0
+  ); // contributors will see logs in console while on development env
   if (!logger) {
     return;
   }
 
+  if (botConfigError) {
+    logger.error(botConfigError.toString());
+    return;
+  }
```

Comment:
I guess these lines need to be moved to here.

line 38 - line 42

 ```
 try {
    botConfig = await loadConfig(context);
  } catch (err) {
    botConfigError = err;
  }
```

Reply:
ideally yes, but logger needs to be initialized for this to work, but logger gets initialized after loading the config because it needs config for supabase and other variables. but if the loading of the config fails then we don't have those variables - this is why I had to make additional fallbacks like [here](https://github.com/ubiquity/ubiquibot/pull/684#discussion_r1309506414)

Reply:
oh right.